### PR TITLE
Add SyncSegments into proto

### PR DESCRIPTION
**Pull request by XuanYang-cn**
See also: #19072

Signed-off-by: yangxuan <xuan.yang@zilliz.com>

### CODE REVIEW

**Comment by XuanYang-cn** [comment]
rerun ut

File: internal/datanode/data_node.go [hunk 1]
```diff
@@ -877,6 +877,65 @@ func (node *DataNode) GetCompactionState(ctx context.Context, req *datapb.Compac
 	}, nil
 }
 
+// SyncSegments called by DataCoord, sync the compacted segments' meta between DC and DN
+func (node *DataNode) SyncSegments(ctx context.Context, req *datapb.SyncSegmentsRequest) (*commonpb.Status, error) {
+	log.Info("DataNode receives SyncSegments",
```
Comment (jiaoew1991) [inline internal/datanode/data_node.go:882]: log.Ctx(ctx).Info

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**